lib/sorcery/controller/config.rb

```diff
@@ -54,11 +54,8 @@ def user_config(&blk)
         end
 
         def configure(&blk)
-          @configure_blk = blk
-        end
-
-        def configure!
-          @configure_blk.call(self) if @configure_blk
+          update!
+          blk.call(self) if blk
         end
```

Comment:
With how the config works in v1, I don't know that immediately calling the block makes sense. `configure` essentially sets the global config, which gets executed on duplicates rather than the primary instance. I'll definitely keep this in mind though, and see if it can be simplified to a single method when testing.

Reply:
> `configure` essentially sets the global config, which gets executed on duplicates rather than the primary instance.

While inspecting the code I found out that `Sorcery::Controller::Config` class (not instance) is intended to be used in controllers and it is not relevant where we set class variables (in the `configure!` method after action controller is laded or in the sorcery initializer when calling `configure` with a block)

Instances of `Sorcery::Model::Config` [are configured in `::Sorcery::Controller::Config.user_config` block](https://github.com/Sorcery/sorcery/blob/v0.14.0/lib/sorcery/model.rb#L20) and used in user models

There is another way to fix #227: move configure user block out of the configure controller block. Sorcery initializer will look like this:

```ruby
# Rails.application.config.sorcery.configure is a way too long, 
# Sorcery.configure looks more elegant
Sorcery.configure_controller do |config|
  ...
end

Sorcery.configure_user do |config|
  ...
end

```

Reply:
I definitely like calling `Sorcery.configure` over `Rails.application.config.sorcery.configure`, I'll see about implementing that for sure.

I've reworked how the config works quite a bit in the rework, but if there ends up still being a `configure_user` block, I'll extract it out like you've suggested. I like that separation better.